ui/src/pages/backoffice/Acquisition/Order/OrderEditor/components/OrderForm/OrderForm.js

```diff
@@ -16,22 +16,26 @@ import _has from 'lodash/has';
 const orderSubmitSerializer = (values, newRecord) => {
   const submitValues = { ...values };
   if (newRecord) {
-    submitValues.created_by_pid = sessionManager.user.id;
+    submitValues.created_by_pid = String(sessionManager.user.id);
+  }
+  if (_has(values, 'vendor.pid')) {
+    submitValues.vendor_pid = values.vendor.pid;
+  }
+  if (_has(values, 'resolved_order_lines')) {
+    submitValues.order_lines = values.resolved_order_lines.map(line => {
+      if (line.document) {
+        line.document_pid = _has(line.document, 'id')
+          ? line.document.id
```

Comment:
Can't remember why this `doc.id` and not always `doc.pid`, same for the patron, maybe we can discuss it IRL.